### PR TITLE
Catch missing node error when rotating database

### PR DESCRIPTION
**Pull request by seelabs**
While there should never be a missing node when copying the SHAMap,
rippled should not terminate when there's an error rotating the
database. This patch aborts the database rotation rather than aborting rippled.

- [ x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: src/ripple/app/misc/SHAMapStoreImp.cpp [hunk 1]
```diff
@@ -363,11 +364,24 @@ SHAMapStoreImp::run()
...
+                    std::bind(
+                        &SHAMapStoreImp::copyNode,
+                        this,
+                        std::ref(nodeCount),
+                        std::placeholders::_1));
+            }
+            catch (SHAMapMissingNode const& e)
+            {
+                JLOG(journal_.error())
+                    << "Missing node while copying ledger before roate: "
```
Comment (mtrippled) [inline src/ripple/app/misc/SHAMapStoreImp.cpp:380]: spell check "rotate"

**Review by mtrippled — APPROVED** [review]
LGTM :+1: but fix the spelling of "rotate"

File: src/ripple/app/misc/SHAMapStoreImp.cpp (hunk 1 above)
Comment (seelabs) [reply src/ripple/app/misc/SHAMapStoreImp.cpp]: Fixed.

File: src/ripple/app/misc/SHAMapStoreImp.cpp [hunk 2]
```diff
@@ -30,6 +30,7 @@
 #include <ripple/nodestore/Scheduler.h>
 
 #include <boost/algorithm/string/predicate.hpp>
+#include "ripple/shamap/SHAMapMissingNode.h"
```
Comment (scottschurr) [inline src/ripple/app/misc/SHAMapStoreImp.cpp:33]: Could you use chevrons to enclose this `#include`?  And move it up with the `ripple` includes please?  Thanks.

**Review by scottschurr — COMMENTED** [review]
Changes look good to me once the new `#include` is fixed up.  Thanks.

I'm going to let this code burn in for a while on the same machine that saw the crash yesterday.  There's no guarantee that I'll see the same problem, but I may as well give it a try.  If I don't see any crashes after it burns in for a few hours I'll give it the thumbs up.

File: src/ripple/app/misc/SHAMapStoreImp.cpp (hunk 2 above)
Comment (seelabs) [reply src/ripple/app/misc/SHAMapStoreImp.cpp]: @scottschurr Fixed - also an explanation: I changed my editor configuration and it started "helpfully" adding include files for me. I need to figure out how to turn that off. I missed that is added that for me. Thanks for spotting it!

Comment (haihai423) [reply src/ripple/app/misc/SHAMapStoreImp.cpp]: 你们和雷达币有关吗

**Review by scottschurr — APPROVED** [review]
I did a burn in with this code over the weekend.  Unfortunately, and to no one's surprise, the burning in code neither crashed nor left a message in the log to verify that we really fixed things up with this change.  But I think there's a high likelihood that this is the right change.

Let's do this.  Thanks for the fix.  :+1:

File: src/ripple/app/misc/SHAMapStoreImp.cpp (hunk 2 above)
Comment (nbougalis) [reply src/ripple/app/misc/SHAMapStoreImp.cpp]: @haihai423: Short answer: no. Longer answer: I believe that Radar Coin simply cloned this code (along with several libraries and even entire websites), only changing the name.

**Comment by seelabs** [comment]
Squashed and marked passed

**Pull request closed without merging**